tsdb/head.go

```diff
@@ -1648,25 +1673,33 @@ func (s *stripeSeries) getByHash(hash uint64, lset labels.Labels) *memSeries {
 	return series
 }
 
-func (s *stripeSeries) getOrSet(hash uint64, series *memSeries) (*memSeries, bool) {
-	i := hash & uint64(s.size-1)
+func (s *stripeSeries) getOrSet(hash uint64, series *memSeries) (*memSeries, bool, error) {
+	createSeriesErr := s.seriesLifecycleCallback.PreCreationCallback(series.lset)
```

Comment:
This is unexpected to me. I would expect preCreation callback to invoked before creation. That moment is after 

```
if prev := s.hashes[i].get(hash, series.lset); prev != nil {
		s.locks[i].Unlock()
		return prev, false, nil
	}
```
I think... no?

Reply:
We don't want to have the callback inside the lock. Hence `PreCreation` is just a hint to TSDB about what to do if it had to create a series and it is a best-effort as we want to avoid callbacks inside lock.

Reply:
Interesting.... I think this make sense, but deserve a comment.. WDYT? (:

Reply:
Definitely